CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+elseif(APPLE AND CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set(HELICS_DOWNLOAD_URL https://github.com/GMLC-TDC/HELICS/releases/download/v${SKBUILD_PROJECT_VERSION}/Helics-${SKBUILD_PROJECT_VERSION}-macOS-universal2.zip)
+elseif(WIN32)
+    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+        set(HELICS_DOWNLOAD_URL https://github.com/GMLC-TDC/HELICS/releases/download/v${SKBUILD_PROJECT_VERSION}/Helics-${SKBUILD_PROJECT_VERSION}-win64.zip)
+    elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
+        set(HELICS_DOWNLOAD_URL https://github.com/GMLC-TDC/HELICS/releases/download/v${SKBUILD_PROJECT_VERSION}/Helics-${SKBUILD_PROJECT_VERSION}-win32.zip)
+    endif()
+endif()
+
+# If we are building a binary wheel for an official release (or installing) and CIBUILDWHEEL is set, speed things up with pre-built HELICS binaries
+if((DEFINED ENV{CIBUILDWHEEL} OR DEFINED ENV{DOWNLOAD_BINARIES}) AND HELICS_DOWNLOAD_URL AND SKBUILD_STATE STREQUAL "wheel")
+    message(STATUS "Downloading pre-built HELICS release")
+    FetchContent_Declare(
+        helics-bin
+        URL ${HELICS_DOWNLOAD_URL}
+    )
+    FetchContent_MakeAvailable(helics-bin)
+    # Only include the things we actually need for the helics_apps in the wheels to minimize overall file size
+    include(GNUInstallDirs) # used so default BIN/LIB subfolder names will match HELICS
+    if(EXISTS ${helics-bin_SOURCE_DIR}/lib)
+        install(
+            DIRECTORY ${helics-bin_SOURCE_DIR}/lib/
+            TYPE LIB
+            USE_SOURCE_PERMISSIONS
+            PATTERN *
+        )
+    endif()
+    if(EXISTS ${helics-bin_SOURCE_DIR}/lib64)
+        install(
+            DIRECTORY ${helics-bin_SOURCE_DIR}/lib64/
+            TYPE LIB
+            USE_SOURCE_PERMISSIONS
+            PATTERN *
+        )
+    endif()
+    install(
+        DIRECTORY ${helics-bin_SOURCE_DIR}/bin/
+        TYPE BIN
+        USE_SOURCE_PERMISSIONS
+        PATTERN *
+    )
+    install(
+        DIRECTORY ${helics-bin_SOURCE_DIR}/include/
+        TYPE INCLUDE
+        USE_SOURCE_PERMISSIONS
+        PATTERN *
+    )
+    install(
+        DIRECTORY ${helics-bin_SOURCE_DIR}/share/
+        TYPE DATA
+        USE_SOURCE_PERMISSIONS
+        PATTERN *
+    )
+
+    # If we made a wheel from pre-built HELICS binaries, exit early!
+    return()
+endif()
+
+# No pre-built HELICS binaries being used, warn about limitations of building wheels or installing from an sdist
+if(SKBUILD_STATE STREQUAL "wheel")
+    message(WARNING "Building HELICS from source (some features including ZeroMQ Core, IPC Core, and web server won't be available)")
+endif()
+
+set(CMAKE_CXX_STANDARD 17)
+set(HELICS_DISABLE_GIT_OPERATIONS ON)
+set(HELICS_ZMQ_SUBPROJECT ON)
+set(HELICS_ZMQ_FORCE_SUBPROJECT ON)
+set(HELICS_ENABLE_ZMQ_CORE ON) # TODO: vendor libzmq similar to other 3rd party libraries
+set(HELICS_DISABLE_BOOST OFF) # TODO: headers are just needed to compile, maybe tell users to have it installed as prereq for building from source?
```

Comment:
I think we should users to have it installed as a pre-requisite for buidling from source. I think we need to update the pyproject.toml file to exclude the copy of boost that was downloaded (or have cibuildwheel on Linux stick it in a folder outside of the repository clone). A copy of Boost has made its way into the current source dist, making it take up about 250MB compressed -- or 1+ GB uncompressed.

Reply:
I've moved it to /tmp/boost. I'm still not sure why it got included in the sdist.